src/docfx/build/page/BuildPage.cs

```diff
@@ -313,10 +312,21 @@ private static (TemplateModel model, JObject metadata) CreateTemplateModel(Conte
             return (model, metadata);
         }
 
+        private static string CreateHtmlContent(Context context, HtmlNode html, Document file, HashSet<string> bookmarks = null)
+        {
+            // add bookmark validation
+            bookmarks = bookmarks ?? HtmlUtility.GetBookmarks(html);
+            context.BookmarkValidator.AddBookmarks(file, bookmarks);
+
+            // get conceptual from htmldom
+            return LocalizationUtility.AddLeftToRightMarker(file.Docset.Culture, HtmlUtility.AddLinkType(html, file.Docset.Locale).WriteTo());
```

Comment:
if we can make `AddLeftToRightMarker` return an `HtmlNode`, than it could be chained with `StripTags` and we don't need a `CreateHtmlContent` method

Reply:
What about AddLinkType?

Reply:
yep, missed that